### DISCUSSION TITLE
features/monitoring

### DISCUSSION THREAD

**Discussion by giscus** [graphql-hive/console#1442] [Docs Discussions]
# features/monitoring

GraphQL Hive Documentation

https://docs.graphql-hive.com/features/monitoring

**Comment by yuhanz** [comment]
What is the overhead of enabling operation monitoring? Are the access logs transmitted to GraphQL-hive server through HTTP calls? I imagine the requests may be sampled or batched, and wonder if this feature increase memory usage, network bandwidth, connections, etc.

**Reply by dotansimha** [reply]
Hi @yuhanz ! 
You are correct, we are batching and optimizing the storage, and then shipping it together over HTTP. You can find the code here: https://github.com/kamilkisiela/graphql-hive/tree/main/packages/libraries/client 

Memory increasing depends on your configuration, you can change the `max` config field and change the sampling count, and adjust the data stored in memory. We found the default configuration as working pretty well for small and large projects.

**Comment by pierrebiver** [comment]
I love your product :) and I know doc can be hard to write, but I actually stumble upon the `router` alternative by hazard when trying to self host it. I did not know it was a possibility (I went with Apollo server) it would be nice to make it somewhat more visible on the self hosting part maybe ?

**Reply by pierrebiver** [reply]
I meant like having in the self hosting and get started section a mention to the possibilities on how to implement the client ? I just followed the video tutorial.

**Reply by pierrebiver** [reply]
also, what kind of permission the token requires in the example above ?
